governing-documents/security.md

```diff
@@ -0,0 +1,268 @@
+knowledge (e.g. in a particular area of cryptography) who is capable of
+fixing a problem in a short period of time.
+1. Each security team member will be a member of the Hyperledger security email list
+(security@lists.hyperledger.org), and, in general, any Hyperledger-wide
+security infrastructure.
+2. If a project has specialized code related to certain aspects of security
+or cryptography (e.g. consensus algorithms or cryptographic algorithms), then
+a corresponding specialist should be on the response team (e.g. someone
+knowledgeable in consensus or cryptography, respectively).  If a specialist
+is not on the team, then the individual who is responsible for contacting or
+engaging the specialists should be designated in their stead.  We emphasize
+that projects should have access to specialists in an area for which they
+maintain code while recognizing that it may not be practical for these experts
+to be on the response team.
+
+The current response team maintainers are as follows:
+
+| Name | Email ID | Chat ID | Area/Specialty |
+|--- | --- | --- | ---|
+| Satoshi Nakamoto | satoshi@nsa.gov | theREALsatoshi | Everything |
+| Hart Montgomery | hart@donotmail.com | uselessIDIOT | Nothing |
+| <> | <> | <> | (if applicable) |
+
+**Discussion forum:**
+
+1. Projects may set up private discussion forums for approved maintainers and
+security team participants to discuss vulnerabilities.
+    1. The project uses the following private discussion forums for approved
+    maintainers and security team participants, which may include discord or
+    email lists:
+        1. <>
+        2. <>
+
+Forums may be private but the existence of a forum must be public.
+
+## Report Intakes
+While all Hyperledger Foundation projects are required to respond to security
+vulnerability disclosures sent to the Hyperledger security list
+(security@lists.hyperledger.org), many projects have other authorized ways
+that reports are taken in as well.
+
+This project has the following recommended ways to submit security
+vulnerabilities.  While the security team members will do their best to
+respond to bugs disclosed in all possible ways, it is encouraged for bug
+finders to report through the following approved channels:
+
+- The Hyperledger Foundation security list:  security@lists.hyperledger.org.
+
+To report a security issue, please send an email with the name of the project,
+a description of the issue, the steps you took to create the issue, affected
+versions, and if known, mitigations for the issue.  Note that this is a
+mandatory response channel.  We encourage reporters to default to this channel
+for reporting vulnerabilities if there is no compelling reason to use other
+channels. Github security vulnerability reporting (see below for more github
+information):  https://docs.github.com/en/code-security/security-advisories/guidance-on-reporting-and-writing/privately-reporting-a-security-vulnerability.
```

Comment:
OpenSSF advise at https://github.com/ossf/oss-vulnerability-guide/blob/main/maintainer-guide.md:
"If your project is on GitHub, we recommend enabling privately reporting a security vulnerability"

It appears that most if not all Hyperledger projects have private reporting enabled.

With private reporting enabled, any public user can report a security vulnerability directly to the project:
https://docs.github.com/en/code-security/security-advisories/guidance-on-reporting-and-writing/privately-reporting-a-security-vulnerability

I therefore think we need some more clarification in this section. Should users "Report a security vulnerability" via GitHub, or should they use the email? Which method is preferred and why?

I think the deciding factor will be who we want to see the initial reports. Do we want it to be the specific project maintainers? Or do we want it to be a cross-project Hyperledger security team? If the former we should guide towards the GitHub security advisory reporting. If the latter we should guide towards the email. Regardless of the decision here, we should make this document clear about the preferred method. (Note, I am open to either approach).

Reply:
I think we need to let the individual projects decide.  Either one would be fine as long as it is clearly specified.  Is there a reason why it should be one or the other?

Besu, for instance, has a very complicated security intake system (because they get bugs reported through a bunch of Ethereum channels).

Reply:
I should also note:  the OpenSSF vulnerability disclosure WG did review this document and didn't have a problem with this part of the text.